src/CamOverlayAPI.ts

```diff
@@ -95,26 +114,85 @@ export class CamOverlayAPI {
+    }
+
+    async getServices(): Promise<TService[]> {
+        const serviceList = await this.get('/local/camoverlay/api/services.cgi?action=get');
+        return serviceList.services;
+    }
+
+    async updateSingleService(serviceId: number, serviceJson: TService): Promise<void> {
+        const path = '/local/camoverlay/api/services.cgi';
+        await this.post(path, JSON.stringify(serviceJson), {
+            action: 'set',
+            service_id: serviceId.toString(),
+        });
+    }
+
     async updateServices(servicesJson: TServiceList) {
         const path = '/local/camoverlay/api/services.cgi?action=set';
         const res = await this.client.post(path, JSON.stringify(servicesJson));
 
         if (!res.ok) {
-            throw new Error(JSON.stringify(res));
+            throw new Error(await responseStringify(res));
+        }
+    }
+
+    //   ----------------------------------------
+    //               Custom Graphics
+    //   ----------------------------------------
+
+    updateCGText(serviceID: number, fields: TField[]) {
+        let field_specs = '';
+
+        for (const field of fields) {
+            const name = field.field_name;
+            field_specs += `&${name}=${field.text}`;
+            if (field.color !== undefined) {
+                field_specs += `&${name}_color=${field.color}`;
+            }
         }
+
+        return this.promiseCGUpdate(serviceID, 'update_text', field_specs);
+    }
+
+    updateCGImagePos(serviceID: number, coordinates: TCoordinates = '', x = 0, y = 0) {
+        const coord = this.formCoordinates(coordinates, x, y);
+        return this.promiseCGUpdate(serviceID, 'update_image', coord);
     }
 
-    updateCGImageFromData(serviceID: number, imageType: ImageType, imageData: Buffer, coordinates = '', x = 0, y = 0) {
+    updateCGImage(serviceID: number, path: string, coordinates: TCoordinates = '', x = 0, y = 0) {
+        const coord = this.formCoordinates(coordinates, x, y);
+        const update = `&image=${path}`;
+        return this.promiseCGUpdate(serviceID, 'update_image', update + coord);
+    }
+
+    updateCGImageFromData(
+        serviceID: number,
+        imageType: ImageType,
+        imageData: Buffer,
+        coordinates: TCoordinates = '',
+        x = 0,
+        y = 0
+    ) {
         const coord = this.formCoordinates(coordinates, x, y);
         const contentType = imageType === ImageType.PNG ? 'image/png' : 'image/jpeg';
         return this.promiseCGUpdate(serviceID, 'update_image', coord, contentType, imageData);
     }
 
-    async promiseCGUpdate(serviceID: number, action: string, params: string, contentType?: string, data?: Buffer) {
+    private async promiseCGUpdate(
+        serviceID: number,
+        action: string,
+        params: string,
+        contentType?: string,
+        data?: Buffer
+    ) {
         const path = `/local/camoverlay/api/customGraphics.cgi?action=${action}&service_id=${serviceID}${params}`;
```

Comment:
just a tip maybe pass params as an Record<string, string | number>, know the user has to know that he has to pass params starting with '&' which is strange